libs/vm/src/analyser.cpp

```diff
@@ -37,9 +36,40 @@
 namespace fetch {
 namespace vm {
 
-std::string Analyser::CONSTRUCTOR       = "[Constructor]";
-std::string Analyser::GET_INDEXED_VALUE = "[GetIndexedValue]";
-std::string Analyser::SET_INDEXED_VALUE = "[SetIndexedValue]";
+std::string const Analyser::CONSTRUCTOR       = "[Constructor]";
+std::string const Analyser::GET_INDEXED_VALUE = "[GetIndexedValue]";
+std::string const Analyser::SET_INDEXED_VALUE = "[SetIndexedValue]";
+
+namespace {
+
+std::string const INIT_ANNOTATION      = "@init";
+std::string const ACTION_ANNOTATION    = "@action";
+std::string const QUERY_ANNOTATION     = "@query";
+std::string const PROBLEM_ANNOTATION   = "@problem";
+std::string const OBJECTIVE_ANNOTATION = "@objective";
+std::string const WORK_ANNOTATION      = "@work";
+std::string const CLEAR_ANNOTATION     = "@clear";
+
+bool HasAnnotation(NodePtr const &annotations_node, std::string const &annotation_name)
+{
+  if (annotations_node)
+  {
+    assert(annotations_node->node_kind == NodeKind::Annotations);
+
+    for (auto const &annotation_ptr : annotations_node->children)
```

Comment:
Niggle, but I don't use _ptr in names. Just use "annotation" here?

Reply:
done.